Fronter/Source/UpdateChecker/UpdateChecker.cpp

```diff
@@ -0,0 +1,216 @@
+#include "UpdateChecker.h"
+#include "VersionParser.h"
+#include <set>
+
+
+bool newerVersion(const std::string& currentVersion, const std::set<std::string>& versions)
+{
+	auto newestVersion = currentVersion;
+	auto changed = false;
+	for (const auto& version : versions)
+	{
+		const auto newestDotPos = newestVersion.find('.');
+		const auto versionDotPos = version.find('.');
+		
+		if (std::stoi(version) > std::stoi(newestVersion))
+		{
+			newestVersion = version;
+			changed = true;
+			continue;
+		}
+		if (std::stoi(version) < std::stoi(newestVersion))
+			continue;
+
+		// if parts before dot are equal, compare the rest
+		auto newestRSide = newestDotPos == std::string::npos ? "" : newestVersion.substr(newestDotPos + 1);
+		auto versionRSide = versionDotPos == std::string::npos ? "" : version.substr(versionDotPos + 1);
+		if (version.size() > newestVersion.size())
```

Comment:
You should be able to replace these loops with judicious use of std::transform (see the first example [here](https://en.cppreference.com/w/cpp/algorithm/transform)) and std::lexicographical_compare

Reply:
It might even invalidate the need to compare both parts separately.